data-collection/data-collection/Rich Popup/RichPopupManager.swift

```diff
@@ -395,29 +408,29 @@ class RichPopupManager: AGSPopupManager {
     ///
     private func update(manyToOne popup: AGSPopup, forRelationship info: AGSRelationshipInfo) throws {
         
-        guard let relationships = richPopup.relationships, relationships.loadStatus == .loaded else {
-            throw richPopup.relationships?.loadError ?? NSError.unknown
+        guard let relationships = richPopup.relationships,
+              relationships.loadStatus == .loaded
+        else {
+            throw richPopup.relationships?.loadError ?? InvalidOperation()
         }
         
         guard isEditing else {
-            throw NSError.invalidOperation
+            throw InvalidOperation()
         }
         
         guard !info.isComposite else {
-            throw RichPopupManagerError.cannotRelateFeatures
+            // Data Collection does not support non-composite many to one relationships at this time.
+            throw InvalidOperation()
         }
         
-        let foundManager = relationships.manyToOne.first { (manager) -> Bool in
-            
-            guard let relationshipInfo = manager.relationshipInfo else {
-                return false
+        let foundManager = relationships.manyToOne
+            .first { (manager) -> Bool in
+                guard let relationshipInfo = manager.relationshipInfo else { return false }
+                return relationshipInfo == info
             }
```

Comment:
This is more complicated than it needs to be. An optional value can be compared to a non-optional one using [`==`](https://developer.apple.com/documentation/swift/optional/2950146):

> Use this equal-to operator (==) to compare any two optional instances of a type that conforms to the Equatable protocol. The comparison returns true if both arguments are nil or if the two arguments wrap values that are equal. Conversely, the comparison returns false if only one of the arguments is nil or if the two arguments wrap values that are not equal.

```suggestion
            .first { $0.relationshipInfo == info }
```